lib/Dialect/FIRRTL/FIRRTLTypes.cpp

```diff
@@ -822,14 +828,16 @@ bool firrtl::areTypesEquivalent(FIRRTLType destFType, FIRRTLType srcFType,
     return srcType.isResetType();
 
   // Reset types can drive UInt<1>, AsyncReset, or Reset types.
-  if (srcType.isa<ResetType>())
+  if (srcType.isa<ResetType>() && !requireDestSameOrUninferred)
```

Comment:
Do you not want to support casting from `probe<asyncreset>` to `probe<reset>`?

Reply:
This is meant to block going from `reset` to a more narrow type (`asyncreset`), going the other way is fine.

Reply:
Ah, got it